components/reddit/reddit.app.js

```diff
@@ -0,0 +1,254 @@
+      type: "string",
+      label: "Username",
+      description: "The username you'd like to watch.",
+    },
+    timeFilter: {
+      type: "string",
+      label: "Time filter",
+      description:
+        "If set to all, all existing links or comments, before applying dedupe strategy, will be considered to be emitted. Otherwise, the indicated time frame will be used for getting links or comments.",
+      options: ["hour", "day", "week", "month", "year", "all"],
+      default: "all",
+      optional: true,
+    },
+    includeSubredditDetails: {
+      type: "boolean",
+      label: "Include subreddit details?",
+      description:
+        "If set to true, subreddit details will be expanded/included in the emitted event.",
+      default: false,
+      optional: true,
+    },
+  },
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    _isRetriableStatusCode(statusCode) {
+      [408, 429, 500].includes(statusCode);
+    },
+    async _withRetries(apiCall) {
+      const retryOpts = {
+        retries: 5,
+        factor: 2,
+      };
+      return retry(async (bail) => {
+        try {
+          return await apiCall();
+        } catch (err) {
+          const statusCode = get(err, ["response", "status"]);
+          if (!this._isRetriableStatusCode(statusCode)) {
+            bail(`
+              Unexpected error (status code: ${statusCode}):
+              ${err.response}
+            `);
+          }
+          console.warn(`Temporary error: ${err.message}`);
+          throw err;
+        }
+      }, retryOpts);
+    },
+    /**
+     * This method retrieves the most recent new hot subreddit posts. The
+     * returned dataset contains at most `opts.limit` entries.
+     *
+     * @param {object}  opts options to customise the data retrieval
+     * @param {string}  opts.subreddit the subreddit from which to retrieve the
+     * hot posts
+     * @param {enum}    opts.region the region from where to retrieve the hot
+     * posts (e.g. `GLOBAL`, `US`, `AR`, etc.). See the `g` parameter in the
+     * docs for more information: https://www.reddit.com/dev/api/#GET_hot
+     * @param {boolean} [opts.excludeFilters=false] if set to `true`, filters
+     * such as "hide links that I have voted on" will be disabled
+     * @param {boolean} [opts.includeSubredditDetails=false] whether the
+     * subreddit details should be expanded/included or not
+     * @param {number}  [opts.limit=100] the maximum amount of posts to retrieve
+     */
```

Comment:
**-Minor-**

The docs dont' exactly match the signature of the method. The docs say that the method takes a single object (named `opts`) and that the object contains attributes such as `opts.subreddit`, `opts.region`, etc.

This method on the other hand takes those attributes as separate arguments.

This should fix it:
```suggestion
    /**
     * This method retrieves the most recent new hot subreddit posts. The
     * returned dataset contains at most `opts.limit` entries.
     *
     * @param {string}  subreddit the subreddit from which to retrieve the
     * hot posts
     * @param {enum}    region the region from where to retrieve the hot
     * posts (e.g. `GLOBAL`, `US`, `AR`, etc.). See the `g` parameter in the
     * docs for more information: https://www.reddit.com/dev/api/#GET_hot
     * @param {boolean} [excludeFilters=false] if set to `true`, filters
     * such as "hide links that I have voted on" will be disabled
     * @param {boolean} [includeSubredditDetails=false] whether the
     * subreddit details should be expanded/included or not
     * @param {number}  [limit=100] the maximum amount of posts to retrieve
     * @returns the list of new hot posts belonging to the specified subreddit
     */
```

Reply:
Sorry, I didn't give much of a thought to the opts. option in the docs. I have committed the suggestion.